### ISSUE TITLE
Low batch size vs loss >1 in SSD7 training

### ISSUE THREAD

**Issue by mhtefe** [pierluigiferrari/ssd_keras#41]
Hi, thanks a lot for sharing your illuminating work but I have one issue tickles my mind.

When I try to train the model in my laptop, whic does not have good GPU memory, I lowered the batch size to 4 and then I began to see loss values around like 2 during training. Though it gets lower during training, it was a little bit suprise to see values bigger than 1.  

What may cause this behaviour ? Should I take a look at 'SSDLoss'

Kind Regards

**Comment by pierluigiferrari** [comment]
Oh yeah, you could totally see loss values greater than 1. Much greater than 1 in the first few training steps when training from scratch, actually. What makes you think that this shouldn't be the case?

Also, are you using a recent commit? I've actually made the loss value independent of the batch size a few commits ago, so the loss should always start out and develop relatively similarly regardless of the batch size.

**Comment by mhtefe** [comment]
I just thought that normalization of total_loss ( at SSDLoss function in keras_ssd_loss.py ) should have prevented to see values grater than 1 ( I think last multiplication operation, which prevents keras batch averaging, may cause it ). That's not a big deal as long as it decreases.

Thanks for sharing 👍 👍 👍

**Comment by pierluigiferrari** [comment]
I think you might have a general misconception about what the loss can look like. Loss is not like accuracy or precision where you have a value that ranges between zero and one. The loss for even one box, which is composed of the log loss for the box's class and the smooth L1 loss for the box's coordinates, can already be much larger than one. The log loss alone goes to infinity as the confidence goes towards zero. The lowest likely confidences for and untrained network depend on the number of classes of course, but the more classes you have, the higher the log loss will start out at. This is regardless of whether you average across the number of boxes or the like.
